onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -1450,31 +1460,38 @@ void ConvOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const Nod
   input_indices.push_back(model_builder.AddOperandFromScalar(onnx_strides[1]));
   input_indices.push_back(model_builder.AddOperandFromScalar(onnx_strides[0]));
 
-  if (!conv2d && depthwise_conv2d) {
-    int32_t depthwiseMultiplier = shaper[weight][3] / group;
-    input_indices.push_back(model_builder.AddOperandFromScalar(depthwiseMultiplier));
+  if (!conv_2d) {
+    if (depthwise_conv_2d) {
+      int32_t depthwiseMultiplier = shaper[weight][3] / group;
+      input_indices.push_back(model_builder.AddOperandFromScalar(depthwiseMultiplier));
+    } else {  // grouped_conv_2d
+      input_indices.push_back(model_builder.AddOperandFromScalar(group));
+    }
   }
 
```

Comment:
I'm not a Conv expert so hard for me to judge correctness here. I assume the existing unit tests cover it. Are you able to run a code coverage tool on Android to validate that the various branches in these changes are covered?

Reply:
Yes, all the test passed, I also added case for grouped conv

---
In reply to: [459173161](https://github.com/microsoft/onnxruntime/pull/4582#discussion_r459173161) [](ancestors = 459173161)